### ISSUE TITLE
Database is locked

### ISSUE THREAD

**Issue by LukasKorba** [zcash/ZcashLightClientKit#1281]
SQLite.swift package crashes in failableNext() call. This happens occasionally when transaction history screen is calling allTransactions of the SDKSynchronizer. I suspect a race condition/concurrency issue because the SDKSynchonizer is syncing and at that time it's also supposed to fetch the transactions while probably already accessing the DB.

```swift
extension FailableIterator {
    public func next() -> Element? {
        // swiftlint:disable:next force_try
        try! failableNext()
    }
}
```

**Comment by daira** [comment]
It's not out of our control, since the iterator's `failableNext` method can be called directly from our code as explained in https://github.com/stephencelis/SQLite.swift/issues/1028 and https://github.com/stephencelis/SQLite.swift/pull/1075 .

If there is a concurrency bug, that needs to be fixed independently: the database should only be accessed from one thread.

However, using `failableNext` is the right thing anyway to do correct error reporting and cleanup. (At the very least, this error [can occur](https://github.com/stephencelis/SQLite.swift/issues/1028#issuecomment-751448124)) with access to the db from two processes, even if each one only accesses it from a single thread.)

**Comment by nuttycom** [comment]
Fixed in #1282
